container_pipeline/workers/test.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python
+
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class TestWorker(BaseWorker):
+    """Test Worker"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(TestWorker, self).__init__(logger=logger, sub=sub, pub=pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def test(self, job):
+        """Runs openshift test job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'test')
+            if not build_id:
+                return False
+        except OpenshiftError as e:
+            logger.error(e)
+            return False
```

Comment:
@navidshaikh i believe this is already taken care of by @rtnpro in Openshift class